framework/src/actions/DetermineSystemType.C

```diff
@@ -37,4 +39,27 @@ DetermineSystemType::act()
    */
   if (_moose_object_pars.isParamValid("_eigen") && _moose_object_pars.get<bool>("_eigen"))
     _app.useNonlinear() = false;
+
+  if (_moose_object_pars.isParamValid("_use_eigen_value") &&
+      _moose_object_pars.get<bool>("_use_eigen_value"))
+  {
+    _app.useEigenvalue() = true;
+    _action_params.set<std::string>("type") = "EigenProblem";
+  }
+  else
+  {
+    _action_params.set<std::string>("type") = "FEProblem";
+  }
+
+  // if we have Problem block in input file, "CreateProblemAction" will be handled by parser
+  if (_awh.hasActions("create_problem"))
+    return;
+
+  // Create "CreateProblemAction"
+  std::shared_ptr<MooseObjectAction> action =
```

Comment:
No point to casting. You don't use the casted value. Just create and add instead.